### PR TITLE
[Docs] Index of Partials

### PR DESCRIPTION
**Pull request by alexfornuto**
This creates a non-listed page with all the partial files (includes) in a single index, for Docs contributors to reference for reusable content.
Note that some partials are referenced but not expanded due to https://github.com/gravitational/docs/issues/167.

### CODE REVIEW

**Comment by ptgott** [comment]
I'm taking a look at https://github.com/gravitational/docs/issues/167 this week

**Comment by ptgott** [comment]
The partial issues should be fixed once we merge https://github.com/gravitational/docs/pull/180

**Comment by alexfornuto** [comment]
> The partial issues should be fixed once we merge [gravitational/docs#180](https://github.com/gravitational/docs/pull/180)

I've tested locally using this branch for the submodule and `paul.gottschling/167-png-paths` for the docs repo branch, and it doesn't seem to resolve the issue.

**Comment by ptgott** [comment]
Are you using the latest `paul.gottschling/167-png-paths` branch? If `gravitational/docs` uses that branch and `teleport` uses the latest `afornuto/partials-index`, `/docs/partials` loads without issue for me 🤷

**Comment by alexfornuto** [comment]
@ptgott I must not have had the latest commits from your branch. https://github.com/gravitational/docs/pull/180 FTW!

**Comment by ptgott** [comment]
@alexfornuto Just merged gravitational/docs#180!

**Comment by alexfornuto** [comment]
> @alexfornuto Just merged [gravitational/docs#180](https://github.com/gravitational/docs/pull/180)!

Confirmed that this page is working and ready for review.

File: docs/pages/partials.mdx [hunk 1]
```diff
@@ -0,0 +1,431 @@
+---
+title: Index of Partial Files
```
Comment (ptgott) [inline docs/pages/partials.mdx:2]: I think we can add this to the "Contribute" section of the docs. It's useful for anyone who wants to write a docs page to refer to.

File: docs/pages/partials.mdx [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+description: For tech writers to easily see what content is available for reuse
```
Comment (ptgott) [inline docs/pages/partials.mdx:3]:
```suggestion
description: For documentation contributors to easily see what content is available for reuse.
```

File: docs/pages/partials.mdx [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+layout: tocless-doc
+---
+
+This page indexes the partial files (includes) that are re-used throughout the docs. The include is within the expandable `<Details>` block under each header. Some partials are only referenced but not directly included, because they include local relative links that do not work from the directory path where this index lives.
```
Comment (ptgott) [inline docs/pages/partials.mdx:7]:
```suggestion
This page indexes the partial files (includes) that are re-used throughout the docs. Each include is within the expandable `<Details>` block under each header. 
```

File: docs/pages/partials.mdx [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+
+(!docs/pages/includes/cloud/call-to-action.mdx!)
+
+</Details>
+
+### cloud\/cloudmanagedadvisory.mdx
+
+<Details title="cloud/cloudmanagedadvisory.mdx">
+
+(!docs/pages/includes/cloud/cloudmanagedadvisory.mdx!)
```
Comment (ptgott) [inline docs/pages/partials.mdx:23]: This one is hidden due to scope—should we add `forScopes` in `config.json` for this page?

**Review by ptgott — COMMENTED** [review]
- I think it would make this easier to navigate if we were to alphabetize the filenames
- Do we have a script we can use to generate this page automatically? It'd be good to have a way to keep it up to date as we add/remove partials.

**Comment by alexfornuto** [comment]
@ptgott I'm inclined to abandon work on this PR, per my last update.

**Comment by ptgott** [comment]
@alexfornuto This is a different issue this time, since the page is rendering but failing the linter. I'm guessing this issue isn't limited to the partials index, so we'll need to fix this in general! I'll make time to create an issue

**Comment by ptgott** [comment]
@alexfornuto It turns out that the `gravitational/docs` container image we are using for linting in our CI pipeline dates back to September because builds for the image have been failing, so once we fix that (hopefully today), the linter will accept the image links.

**Comment by alexfornuto** [comment]
@ptgott the time this PR has spent going stale shows that this is a bad idea. We've created new partials since then, and I don't expect us to remember to update this page when we do or periodically check and refresh this page.

I still like the concept, but I think it should be shelved until we can create one programmatically that's always up-to-date.

**Comment by ptgott** [comment]
@alexfornuto That works for me! I also think this would be really helpful in an auto-generated form—maybe we should turn this into an issue?

**Comment by alexfornuto** [comment]
I've drafted a card in the Docs project to track the idea of auto-generating a list of partials. Farewell, old PR.

**Pull request closed without merging**